dataset/dataset.py

```diff
@@ -1,31 +1,37 @@
 import copy
 import gzip
+import json
 import torch
-from torch.utils.data import Dataset, DataLoader, IterableDataset
-from transformers import BertTokenizer
+import math
+from torch.utils.data import Dataset, IterableDataset
+from numpy.random import choice
 
 """
 Implementation of lazy dataloder in PyTorch
 """
 
 class TextIterator:
-  def __init__(self, text_iterator, batch_size, num_workers, transform=None):
+  def __init__(self, text_iterator, batch_size, lengths, weights, num_workers,
+               transform=None):
     self.batch_size = batch_size
     self.iter_number = 0
```

Comment:
I think you should keep a track of the number of iterations performed on each dataset to avoid overlap between multiple workers. Something like self.iter_number = [0 for _ in range(lengths)]